diem-move/diem-transactional-test-harness/src/diem_test_harness.rs

```diff
@@ -830,11 +994,18 @@ impl<'a> MoveTestAdapter<'a> for DiemTestAdapter<'a> {
         view_resource_in_move_storage(&self.storage, address, module, resource, type_args)
     }
 
-    fn handle_subcommand(
-        &mut self,
-        _subcommand: TaskInput<Self::Subcommand>,
-    ) -> Result<Option<String>> {
-        unreachable!()
+    fn handle_subcommand(&mut self, input: TaskInput<Self::Subcommand>) -> Result<Option<String>> {
+        match input.command {
+            DiemSubCommand::BlockCommand(block_cmd) => {
+                let proposer = self.compiled_state().resolve_address(&block_cmd.proposer);
+                let metadata =
+                    BlockMetadata::new(HashValue::zero(), 0, block_cmd.time, vec![], proposer);
+
+                let output = self.run_transaction(Transaction::BlockMetadata(metadata))?;
```

Comment:
Didn't realize this was truly its own transaction type. Seems like a good use case for a subcommand then